azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -210,28 +213,64 @@ private void createLogger() {
               + this.jobId;
       logger = Logger.getLogger(loggerName);
 
-      // Create file appender
-      String logName = createLogFileName(node);
-      logFile = new File(workingDir, logName);
-
-      String absolutePath = logFile.getAbsolutePath();
+      // Attempt to create appenders as necessary
+      attachFileAppender();
 
-      jobAppender = null;
-      try {
-        RollingFileAppender fileAppender =
-            new RollingFileAppender(loggerLayout, absolutePath, true);
-        fileAppender.setMaxBackupIndex(jobLogBackupIndex);
-        fileAppender.setMaxFileSize(jobLogChunkSize);
-        jobAppender = fileAppender;
-        logger.addAppender(jobAppender);
-        logger.setAdditivity(false);
-      } catch (IOException e) {
-        flowLogger.error("Could not open log file in " + workingDir
-            + " for job " + this.jobId, e);
+      if (props.getBoolean("azkaban.job.kafkalogging.enabled", false)) {
+        attachKafkaAppender();
       }
     }
   }
 
+  private void attachFileAppender() {
+    jobAppender = null;
+    String logName = createLogFileName(node);
+    logFile = new File(workingDir, logName);
+    String absolutePath = logFile.getAbsolutePath();
+
+    try {
+      RollingFileAppender fileAppender =
+          new RollingFileAppender(loggerLayout, absolutePath, true);
+      fileAppender.setMaxBackupIndex(jobLogBackupIndex);
+      fileAppender.setMaxFileSize(jobLogChunkSize);
+      jobAppender = fileAppender;
+      logger.addAppender(jobAppender);
+      logger.setAdditivity(false);
+      flowLogger.info("Created file appender for job " + this.jobId);
+    } catch (IOException e) {
+      flowLogger.error("Could not open log file in " + workingDir
+          + " for job " + this.jobId, e);
+    }
+  }
+
+  private void attachKafkaAppender() {
+    kafkaAppender = null;
+
+    try {
+      KafkaLog4jAppender kafkaProducer = new KafkaLog4jAppender();
+      kafkaProducer.setBrokerList(props.getString("azkaban.job.kafkalogging.brokerlist"));
```

Comment:
How would you envision using log4j2's ability to dynamically adjust logging here? What's the use case?